woocommerce/woocommerce-android
# Navigation 2.4: drop Robolectric from tests

Closes #4618 

Please don't merge yet, this depends on #5199, and we need to make sure we merge any latest changes from develop before merging, to confirm that there is no new test classes that would break. 

### Description
The goal of this PR is to remove Robolectric from our test dependencies.
Just to give some context, we had to add Robolectric when we migrated to Hilt, because we needed a way to initialize the `SavedStateHandle` from the navigation `Args` classes, and because they only supported `Bundles` back then, we wrote our functions to use the `Bundle` of the `Args` class, to create a `SavedStateHandle`.
With Navigation 2.4, the generated `Args` classes has a new `toSavedStateHandle` that makes it possible to generate the SavedStateHandle directly, so we don't need Robolectric anymore.

The migration was simple for most of it, except:
1. For some reason, when using Robolectric, Mockito wasn't reporting the error `UnnecessaryStubbingException`, even when it's needed, so I had to go through some tests to remove any unused stubbings 731b691
2. `NotificationMessageHandler` was using an object class `NotificationUtils` for parsing notification messages, and this latter was using some Android frameworks tools: `Base64` and `JSONObject`, I had to update it to allow swapping the implementations 9a1f2d3

### Testing instructions
1. Confirm the tests are all green (should be OK from the CI)
2. Test some notifications, and confirm there is no regression.

### Images/gif
